apps/entry/views.py

```diff
@@ -49,7 +51,31 @@
 import django_filters
 
 
-class EntryViewSet(viewsets.ModelViewSet):
+class EntrySummaryPaginationMixin(object):
+    def get_paginated_response(self, data):
+        if not self.request.data.get('calculate_summary', self.request.GET.get('calculate_summary', '0')) == '1':
+            return super().get_paginated_response(data)
+        qs = self.filter_queryset(self.get_queryset())
+        total_unique_org_type = qs.values('lead__authors__organization_type_id').annotate(count=models.Count('lead__authors__organization_type_id')).count()
+        total_sources = qs.values('lead__source_id').annotate(count=models.Count('lead__source_id')).count()
+        total_leads = qs.values('lead_id').annotate(count=models.Count('lead_id')).count()
+        summary_data = dict(
+            total_verified_entries=qs.filter(verified=True).count(),
+            total_unverified_entries=qs.filter(verified=False).count(),
+            total_leads=total_leads,
+            total_unique_authors=total_unique_org_type,
+            total_sources=total_sources,
+        )
+        return response.Response(OrderedDict([
+            ('count', self.paginator.count),
+            ('next', self.paginator.get_next_link()),
+            ('previous', self.paginator.get_previous_link()),
+            ('results', data),
+            ('summary', summary_data)
+        ]))
```

Comment:
Can we do it like this?
```suggestion
        return response.Response({
            **super().get_paginated_response(data),
            'summary': summary_data,
        })
```